src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,209 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.sps.classes.Notification;
+import com.google.sps.classes.Utility;
+import java.io.IOException;
+import java.sql.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/**
+* Servlet that handles the fetching and posting of notifications.
+*/
+@WebServlet("/notification")
+public class NotificationServlet extends HttpServlet {
+  /**
+  * Fetch the notifications of a specific user.
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get user's ID from Utility method.
+    int userId = Utility.getUserId();
+    // If the user is signed in then fetch notifications.
+    if (userId != -1) {
+      // If the user is logged in, fetch notifications and convert the ArrayList to JSON
+      // using Utility method.
+      String json = Utility.convertToJsonUsingGson(getNotifications(userId));
+      response.setContentType("application/json;");
+      response.getWriter().println(json); 
```

Comment:
You could save yourself the String declaration:
```suggestion
      response.setContentType("application/json;");
      response.getWriter().println(Utility.convertToJsonUsingGson(getNotifications(userId))); 
```

Reply:
Done!